web/main/models.py

```diff
@@ -1429,6 +1429,29 @@ class Meta:
         unique_together = (("user", "casebook"),)
 
 
+class CasebookTag(models.Model):
+    casebook = models.ForeignKey(
+        "Casebook",
+        on_delete=models.CASCADE,
+    )
+    tag = models.ForeignKey(
+        "CasebookTagName",
+        on_delete=models.CASCADE,
+    )
+
+    class Meta:
+        unique_together = (("tag", "casebook"),)
+
+
+class CasebookTagName(models.Model):
+    slug = models.CharField(max_length=100, blank=True, null=True, unique=True)
```

Comment:
`SlugField` here? ([ref](https://docs.djangoproject.com/en/3.2/ref/models/fields/#slugfield)). Also in generally you probably don't want both `unique=True` and nullable. Typically I think of slug-type fields as being required but [auto-populated](https://docs.djangoproject.com/en/3.2/ref/contrib/admin/#django.contrib.admin.ModelAdmin.prepopulated_fields).

Reply:
Yes, the unique/null issue makes sense, thank you for catching it! Didn't know about either `SlugField` or `prepopulated`, I'll modify this to use those.